MyWeather/Networking/WeatherEndPoint.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  WeatherEndPoint.swift
+//  MyWeather
+//
+//  Created by Luis Gonzalez on 11/4/18.
+//  Copyright © 2018 Luis Gonzalez Jr. All rights reserved.
+//
+
+import Foundation
+
+final class WeatherAPI {
+    
+    let session = URLSession.shared
+    let lat:String
+    let long:String
+    let weatherURL:String
+    var urlDetails:String
+    
+    
+    
+    init(lati:String, longi:String) {
+        self.lat = lati
+        self.long = longi
+        weatherURL = "https://api.openweathermap.org"
+        urlDetails = "/data/2.5/weather?lat=\(self.lat)&lon=\(self.long)&units=imperial&APPID=01dfd7a79576fd7292bef57bfb4c1923"
+    }
+    
+    func setupAPI(oncompletion: @escaping (Weather) -> Void) {
+        guard let url = URL(string: weatherURL + urlDetails) else { return }
+        session.dataTask(with: url) { (data, response, err) in
+            guard let data = data else { return }
+            do {
+                let weather = try
+                                JSONDecoder().decode(Weather.self, from: data)
+                oncompletion(weather)
+            } catch let jsonErr {
+                print(jsonErr)
+            }
+        }.resume()
+    }
+
+}
```

Comment:
This is what your weather api object should look like. It will give you greater flexibility 

```suggestion

import Foundation

final class OpenWeather {
    let session = URLSession.shared
    let baseURL = "https://api.openweathermap.org"

    func getWeather(longitude: String, latitude: String, onComplete: (() -> Void)) {
        let getWeatherURI = "/data/2.5/weather?lat=\(latitude)&lon=\(longitude)&units=imperial&APPID=01dfd7a79576fd7292bef57bfb4c1923"
        guard let url = URL(string: baseURL + getWeatherURI) else { return }

        session.dataTask(with: url) { (data, response, err) in
            guard let data = data else { return }
            do {
                let weather = try JSONDecoder().decode(Weather.self, from: data)
                onComplete(weather)
            } catch let jsonErr {
                print(jsonErr)
            }
        }.resume()
    }
}


```

Reply:
Please review the code above and let me know if it make sense

Reply:
I understand why you changed it. it does make sense. simplifies it

Reply:
✅